projects/ng-components/lib/toggle-selector/toggle-selector.component.html

```diff
@@ -0,0 +1,28 @@
+<ng-container *ngFor="let item of displayList; let idx = index; trackBy: trackByFn">
+    <div
+        class="ac toggle-selector"
+        [class.error]="error"
+        [class.checked]="item.selected"
+        [class.disabled]="disabled || item.optionItem.disabled"
+        [class.focus]="focusedItem === idx"
+        (click)="handleItemOnClick($event, idx)"
+    >
+        <input
+            [type]="inputType"
+            [attr.aria-label]="item.optionItem.label"
+            [id]="item.optionItem.value"
+            [name]="name"
+            [disabled]="disabled || item.optionItem.disabled"
+            (focus)="focus(idx)"
+            (blur)="blur()"
+            [defaultChecked]="item.selected"
+        />
+        <div class="icon-container" *ngIf="item.optionItem.icon" [innerHTML]="item.optionItem.icon | safeHtml"></div>
+        <label [attr.for]="item.optionItem.value" *ngIf="!item.optionItem.customLabel; customLabel">
+            {{ item.optionItem.label }}
+        </label>
+        <div #customLabel [innerHtml]="item.optionItem.customLabel"></div>
+    </div>
```

Comment:
I think there is a chance of `item.optionItem` to not exist. Therefore, you either implement optional chaining in all the instances if `item.optionItem.x`. Or you add an `ngIf` to a container.

Reply:
there is no possibility for it to happen and here is why:
the template you see now is inside a loop that uses an array called disaplyList of type:
```typescript
interface IDisplayToggleSelector extends IToggleSelector {
    optionItem: IToggleSelector;
    selected: boolean;
}

export interface IToggleSelector {
    value: string;
    label?: string;
    icon?: string;
    iconPosition?: IconPosition;
    customLabel?: HTMLElement;
    description?: string;
    disabled?: boolean;
}
```

displayList Array is populated when the @input list of the component is set, there for it always have optionItem object.